src/core/backend/pearl/group.rs

```diff
@@ -319,16 +321,32 @@ impl Group {
                 .into_string()
                 .map_err(|_| warn!("cannot parse file name: {:?}", entry))
             {
-                let start_timestamp = file_name
-                    .parse()
-                    .map_err(|_| warn!("cannot parse file name: {:?} as timestamp", entry))
-                    .expect("parse file name");
-                let pearl_holder = self.create_pearl_holder(start_timestamp);
-                holders.push(pearl_holder);
+                let partition_name = PartitionName::try_from_string(&file_name);
+                if let Some(partition_name) = partition_name {
+                    let pearl_holder = self.create_pearl_holder(partition_name.timestamp);
+                    holders.push(pearl_holder);
+                } else {
+                    warn!("failed to parse partition name from {}", file_name);
+                }
             }
         }
         Ok(holders)
     }
+
+    fn get_node_hash(&self) -> String {
+        let hash = digest(&SHA256, self.node_name.as_bytes());
+        let hex: Vec<u8> = hash
+            .as_ref()
+            .iter()
+            .flat_map(|&byte| {
+                vec![
+                    ((byte >> 4) & 0b01111u8 | 0b01000000) + 1,
```

Comment:
some comments what is happening here would be nice

Reply:
Added comments